WordPress/gutenberg
# Utilize WordPress PHPUnit Test Library Included By `wp-env`

## What?
This makes the appropriate changes to move us away from needing `wp-phpunit/wp-phpunit`. In the process, I also found that `Yoast/PHPUnit-Polyfills` was not being fully utilized. One of the changes was to change methods like `setUp()` to `set_up()` to prevent PHP 8 compatibility issues.

In addition, as per #39863, the `phpunit` container does not work on M1 chips. This pull request replaces the usage of the `phpunit` container with direct usage of `vendor/bin/phpunit` from Gutenberg.

Closes #39863.

## Why?
#41780 made it so that there's a version of WordPress' PHPUnit test files available within the `wp-env` container. We should dogfood this implementation and use the test files here.

## How?
This pull request makes a few changes to methods in order to be compatible with `Yoast/PHPUnit-Polyfills`:
```
* `setUp()` -> `set_up()`
* `tearDown()` -> `tear_down()`
* `setUpBeforeClass()` -> `set_up_before_class()`
* `setUpAfterClass()` -> `set_up_after_class()`
```

This also removes the `wp-phpunit/wp-phpunit` dependency since it's no longer necessary.

We also add the `wptests_` database table prefix as we are running the test suite in the tests-wordpress container rather than the phpunit container. 

## Testing Instructions
1. Make sure CI passes because it's using the same test script.
2. Run `npm run test-unit-php` and verify that it passes.
3. Run `npm run test-unit-php-multisite` and verify that it passes.
4. Play around with different WordPress versions (it looks like we're not compatible as far back as we claim!)
